### PR TITLE
Use expenditure to handle multiple token payments

### PR DESCRIPTION
**Pull request by youngkidwarrior**
Closes #[806]
Implements expenditure for multiple token payments in one-tx-payment extension for #806

See https://github.com/JoinColony/colonyNetwork/pull/817

### CODE REVIEW

**Comment by kronosapiens** [comment]
@youngkidwarrior can you fix the linting errors?

File: contracts/extensions/OneTxPayment.sol [hunk 1]
```diff
@@ -46,48 +46,96 @@ contract OneTxPayment {
...
   {
-    // Check caller is able to call {add,finalize}Payment and moveFundsBetweenPots on the colony in the domain in question
+    //Arrays must be of equal size
+    require(
+      _workers.length == _tokens.length && _workers.length == _amounts.length,
+      "colony-one-tx-payment-arrays-must-be-equal-length"
+    );
+    // Check caller is able to call {add,finalize}Payment and moveFundsBetweenPots on the colony
     validateCallerPermissions(_callerPermissionDomainId, _callerChildSkillIndex, _domainId);
-    // In addition, check the caller is able to call moveFundsBetweenPots from the root domain
```
Comment (kronosapiens) [inline contracts/extensions/OneTxPayment.sol:67]: Can you put this comment back?

File: contracts/extensions/OneTxPayment.sol [hunk 2]
```diff
@@ -46,48 +46,96 @@ contract OneTxPayment {
...
     require(
       ColonyAuthority(colony.authority()).canCall(msg.sender, 1, address(colony), MOVE_FUNDS_SIG),
       "colony-one-tx-payment-root-funding-not-authorized"
     );
+    //Get around stack too deep with array
+    //paymentData[0] = expenditure slot
+    //paymentData[1] = expenditureId/paymentId
+    //paymentData[3] = fundingPotId
```
Comment (kronosapiens) [inline contracts/extensions/OneTxPayment.sol:79]: Space after // in the comments

File: contracts/extensions/OneTxPayment.sol [hunk 3]
```diff
@@ -46,48 +46,96 @@ contract OneTxPayment {
...
+    uint256[3] memory paymentData;
+    paymentData[0] = 0;
+    uint256 index = 0;
```
Comment (kronosapiens) [inline contracts/extensions/OneTxPayment.sol:82]: Can you use a more descriptive name?

File: contracts/extensions/OneTxPayment.sol [hunk 4]
```diff
@@ -46,48 +46,96 @@ contract OneTxPayment {
...
-      1, // Root domain funding pot is always 1
-      payment.fundingPotId,
-      _amount,
-      _token
-    );
-    colony.finalizePayment(_permissionDomainId, _childSkillIndex, paymentId);
-
-    // Claim payout on behalf of the recipient
-    colony.claimPayment(paymentId, _token);
+    for (index = 0; index < _workers.length; index++) {
```
Comment (kronosapiens) [inline contracts/extensions/OneTxPayment.sol:104]: Just go ahead and initialize index here, do `for (uint256 index; ...)`

File: contracts/extensions/OneTxPayment.sol [hunk 5]
```diff
@@ -98,44 +146,92 @@ contract OneTxPayment {
...
+        _childSkillIndex,
+        _workers[0],
+        _tokens[0],
+        _amounts[0],
+        _domainId,
+        _skillId);
+      ColonyDataTypes.Payment memory payment = colony.getPayment(paymentData[1]);
+      paymentData[2] = payment.fundingPotId;
+    }
+    for (index = 0; index < _workers.length; index++) {
```
Comment (kronosapiens) [inline contracts/extensions/OneTxPayment.sol:200]: Initialize `index` here as well.

**Comment by kronosapiens** [comment]
Go ahead and rebase on develop and you'll be able to merge this.

**Comment by youngkidwarrior** [comment]
Thanks! :) no problem. Open source gets tricky especially with Solidity

**Pull request merged**